ghc/Main.hs

```diff
@@ -514,7 +523,12 @@ parseModeFlags args = do
              Nothing     -> doMakeMode
              Just (m, _) -> m
       errs = errs1 ++ map (mkGeneralLocated "on the commandline") errs2
-  when (not (null errs)) $ throwGhcException $ errorsToGhcException errs
+  when (not (null errs)) $ throwGhcException
+#if __GLASGOW_HASKELL__ < 709
+                             $ errorsToGhcException errs
+#else
+                             $ errorsToGhcException $ map (\(L sp e) -> (show sp, e)) errs
```

Comment:
@chrisdone This is questionable - the type `errorsToGhcException` accepts changed from [Located String] to [(String, String)] and the new source calls the fst "Location" and the snd "Error". I'm not sure what a SrcSpan is, though it seems like some source text which may be a location?
https://downloads.haskell.org/~ghc/latest/docs/html/libraries/ghc-7.10.1/src/CmdLineParser.html#errorsToGhcException